src/home/homework-blob.c

```diff
@@ -0,0 +1,236 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later
+ *
+ * Copyright © 2024 GNOME Foundation Inc.
+ *      Original Author: Adrian Vovk
+ */
+
+#include "copy.h"
+#include "fileio.h"
+#include "fd-util.h"
+#include "fs-util.h"
+#include "home-util.h"
+#include "homework-blob.h"
+#include "homework.h"
+#include "install-file.h"
+#include "macro.h"
+#include "path-util.h"
+#include "recurse-dir.h"
+#include "rm-rf.h"
+#include "sha256.h"
+#include "string-util.h"
+#include "tmpfile-util.h"
+#include "umask-util.h"
+#include "utf8.h"
+
+#define BLOB_MAX_SIZE (64U*1024U*1024U)
+
+static int copy_one_blob(
+                int src,
```

Comment:
could you rename this src_fd? we almost always put "fd" in the names of fds, please do so here too